### DISCUSSION TITLE
RtmpStream camera settings

### DISCUSSION THREAD

**Discussion by DenisStadnichenko** [pedroSG94/RootEncoder#1399] [Q&A] [answered]
Hi, thanks for the great library, I'm using RtmpStream to allow me to change the orientation, everything works fine, but I'm having trouble figuring out how to change the camera from main camera to front camera, turn on the flash and set the zoom, can you tell me which direction to go in?

**Comment by pedroSG94** [comment, marked as answer]
Hello,

You need do it using VideoSource like in this example:
https://github.com/pedroSG94/RootEncoder/blob/master/app/src/main/java/com/pedro/streamer/rotation/StreamService.kt#L138
Depend of the video source used (Camera2Source by default) you have different methods:
https://github.com/pedroSG94/RootEncoder/blob/master/library/src/main/java/com/pedro/library/util/sources/video/Camera2Source.kt

If you need extend functionality you can create you own VideoSource and add required methods. For example you can copy that class and add facial recognition method.

**Reply by DenisStadnichenko** [reply]
Thanks for your help, best regards!

**Comment by pedroSG94** [comment]
Closing as solved. You can open other or reopen it if needed

**Reply by DenisStadnichenko** [reply]
Sorry, there is another question, I need to change the video preview settings that in this method rtmpCamera?.prepareVideo(width, height, bitrate, fps, iFrameInterval)

Do you have an example where this is implemented or is it possible at all, I'm not starting the stream, I need to be able to change the settings before starting the stream.

Thanks in advance for the answer, with best regards

**Comment by pedroSG94** [comment]
Hello,

Using RtmpStream prepareVideo method parameters apply to preview, stream and record. You only need stop all (preview, stream and record) before call prepareVideo. You can do the same with prepareAudio

**Reply by DenisStadnichenko** [reply]
Thank you so much! Will definitely check it out now! Best regards!
